source/robot_model/src/Model.cpp

```diff
@@ -124,17 +142,7 @@ state_representation::Jacobian Model::compute_jacobian(const state_representatio
 
 state_representation::Jacobian Model::compute_jacobian(const state_representation::JointPositions& joint_positions,
                                                        const std::string& frame_name) {
-  unsigned int frame_id;
-  if (frame_name.empty()) {
-    // get last frame if none specified
-    frame_id = this->robot_model_.getFrameId(this->robot_model_.frames.back().name);
-  } else {
-    // throw error if specified frame does not exist
-    if (!this->robot_model_.existFrame(frame_name)) {
-      throw (exceptions::FrameNotFoundException(frame_name));
-    }
-    frame_id = this->robot_model_.getFrameId(frame_name);
-  }
+  unsigned int frame_id = get_frame_ids(std::vector<std::string>{frame_name}).back();
```

Comment:
these returns are good cases for `auto`. We don't care too much about the type here since it's just an intermediate variable to be passed on to the next function.